staging/src/k8s.io/apiserver/pkg/cel/mutation/common/val.go

```diff
@@ -104,16 +108,33 @@ func (v *ObjectVal) IsZeroValue() bool {
 // convertField converts a referred ref.Val to its expected type.
 // For objects, the expected type is map[string]any
 // For lists, the expected type is []any
+// For maps, the expected type is map[string]any
 // For anything else, it is converted via value.Value()
-func convertField(value ref.Val) any {
+//
+// It will return an error if the request type is a map but the key
+// is not a string.
+func convertField(value ref.Val) (any, error) {
 	// special handling for lists, where the elements are converted with Value() instead of ConvertToNative
 	// to allow them to become native value of any type.
 	if listOfVal, ok := value.Value().([]ref.Val); ok {
 		var result []any
 		for _, v := range listOfVal {
 			result = append(result, v.Value())
 		}
-		return result
+		return result, nil
+	}
+	// unstructured maps, as seen in annotations
+	// map keys must be strings
+	if mapOfVal, ok := value.Value().(map[ref.Val]ref.Val); ok {
+		result := make(map[string]any)
+		for k, v := range mapOfVal {
+			stringKey, ok := v.Value().(string)
```

Comment:
stringKey should be `k` ,not `v.Value()` here

Reply:
oops. Fixed.